fiona/__init__.py

```diff
@@ -78,16 +80,19 @@ class Path:
     libdir = os.path.join(os.path.dirname(__file__), ".libs")
     os.environ["PATH"] = os.environ["PATH"] + ";" + libdir
 
-from fiona.collection import Collection, BytesCollection
-from fiona.vfs import vsi_path, parse_paths, is_remote
-from fiona._drivers import driver_count, GDALEnv
+from fiona.collection import BytesCollection, Collection
 from fiona.drvsupport import supported_drivers
+from fiona.env import ensure_env_credentialled, Env
+from fiona.errors import FionaDeprecationWarning
+from fiona._env import driver_count, GDALEnv
 from fiona.compat import OrderedDict
 from fiona.io import MemoryFile
 from fiona.ogrext import _bounds, _listlayers, FIELD_TYPES_MAP, _remove, _remove_layer
 from fiona.ogrext import (
     calc_gdal_version_num, get_gdal_version_num, get_gdal_release_name,
     get_gdal_version_tuple)
+from fiona.path import ParsedPath, parse_path, vsi_path
+from fiona.vfs import parse_paths as vfs_parse_paths
```

Comment:
There's a little bit of overlap between these two modules, refactoring to follow in another PR.